src/main/java/seedu/docit/model/appointment/AppointmentContainsPatientPredicate.java

```diff
@@ -0,0 +1,30 @@
+package seedu.docit.model.appointment;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.docit.model.patient.Email;
+
+
+/**
+ * Tests that an {@code Appointment}'s {@code Patient} email matches any of the patient emails given.
+ */
+public class AppointmentContainsPatientPredicate implements Predicate<Appointment> {
+    private final List<Email> emails;
+
+    public AppointmentContainsPatientPredicate(List<Email> emails) {
+        this.emails = emails;
+    }
+
```

Comment:
Why email? Current implementation of patient has name only as the identity field, i.e. Patient are different if name are different, but can have the same email. Might want to change to Name?

Reply:
See `Patient#isSamePatient`

Reply:
Patients can have the same name, I will change the predicate to take in a Patient instead